c/src/v4/string/unterminated.c

```diff
@@ -72,6 +72,18 @@ bool sbp_unterminated_string_set(sbp_string_t *s, size_t maxlen,
   return true;
 }
 
+size_t sbp_unterminated_string_set_truncating(sbp_string_t *s, size_t maxlen,
+                                 const char *new_str, size_t new_str_len) {
+  size_t copied;
+  size_t truncated_len = maxlen > new_str_len ? new_str_len : maxlen;
+  if (!sbp_string_copy_n_to_buf(s->data, &copied, maxlen + 1,
+                              new_str, truncated_len)) {
+    return 0;
+  }
+  s->encoded_len = copied - 1;
+  return s->encoded_len;
+}
+
 bool sbp_unterminated_string_vprintf(sbp_string_t *s, size_t maxlen,
```

Comment:
Can we get variants for the `printf` family as well? I think that would be very helpful when making log messages in particular